classes/i18n/LocaleFile.inc.php

```diff
@@ -82,34 +47,30 @@ function getFilename() {
 	 * @return string
 	 */
 	function translate($key, $params = array(), $locale = null) {
-		if ($this->isValid()) {
-			$key = trim($key);
-			if (empty($key)) {
-				return '';
-			}
+		if (!$this->isValid()) return null;
+		$key = trim($key);
+		if ($key === '') return '';
+
+		$pool = new Stash\Pool(new Stash\Driver\FileSystem(array('path' => Core::getFileCachePath() . '/stash')));
```

Comment:
Are there any circumstances in which `/stash` would be something else?

Reply:
No, but I did want to distinguish the Stash cache directory from the rest of OJS's file caching, in case Stash assumed ownership of *everything* it found there (e.g. removing everything indiscriminately when flushing the cache). However, I did centralize where the configuration was coded so it's not being specified separately every time.

Reply:
Just wondering if there is some way that we can wrap this to avoid typos or people just deciding to put things in their own directory. Maybe:

```php
$pool = new Stash\Pool(new Stash\Driver\FileSystem(['path' => Core::getStashPath()]);
```

Or:

```php
$pool = new Stash\Pool(new Stash\Driver\FileSystem([('path' => Core::getFileCachePath() . '/' . STASH_PATH)]);
```

Reply:
Oh, nevermind, I see you've got `getStashDriver` now. :+1: